### PR TITLE
Improved .gitignore for IntelliJ IDEA

### PR DESCRIPTION
**Pull request by dgiagio**
Ignore IntelliJ IDEA workspace files.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,13 @@
...
 /.settings/
+
+# IntelliJ IDEA
+.idea
+*.iml
+
+# Build and tests
 /bin/
 /dist/
+/test-report/
```
Comment (renner) [inline .gitignore]: There is no test reports produced so far, I would be fine with merging the IntelliJ part though. The test results should be added to .gitignore with a PR that actually generates such results, so please remove it from this patch and add it to another PR accordingly.

**Comment by renner** [comment]
Note that every pull request should have a description. Please add a description to all of your PRs!

**Comment by dgiagio** [comment]
@renner Done, thanks for reviewing.

**Comment by renner** [comment]
Manually merged into master. I adjusted your new patterns a little in order to make them match more precisely, see 5aec108540c852d2c728437024b1e7a145f94e5a. I hope it still works for you, otherwise let me know (couldn't test it since I don't use IDEA).

**Comment by dgiagio** [comment]
It's good, thanks!

**Pull request merged**